src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/HPackEncoder.cs

```diff
@@ -2,26 +2,30 @@
 // Licensed under the Apache License, Version 2.0.
 // See THIRD-PARTY-NOTICES.TXT in the project root for license information.
 
-using System.Collections.Generic;
 using System.Diagnostics;
+#if KESTREL
+using HeadersEnumerator = Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.Http2HeadersEnumerator;
+#else
+using HeadersEnumerator = System.Collections.Generic.IEnumerator<System.Collections.Generic.KeyValuePair<string, string>>;
+#endif
 
 namespace System.Net.Http.HPack
 {
     internal class HPackEncoder
     {
-        private IEnumerator<KeyValuePair<string, string>> _enumerator;
+        private HeadersEnumerator _enumerator;
 
-        public bool BeginEncode(IEnumerable<KeyValuePair<string, string>> headers, Span<byte> buffer, out int length)
+        public bool BeginEncode(HeadersEnumerator enumerator, Span<byte> buffer, out int length)
```

Comment:
I thought the point was that HttpClient didn't need these APIs and they could be removed?

Reply:
HttpClient might not use this code but non-Kestrel code in our repo does, e.g. HttpSys tests, httpcat.

I changed my PR to use an `#if`. It minimizes the amount of overall change.

Reply:
Minimizing changes is not critical at this point, I'd rather have a clean result. If only test code needs these old APIs then is there something straightforward we can do to restructure how they call this?

In the design for dynamic header compression we've already proposed changes that would require moving this enumerator outside of HPackEncoder. Maybe now is a good time to separate them.

Reply:
> If only test code needs these old APIs

Kestrel **and** tests use these APIs. My previous copied and pasted class that wrapped HPackEncoder still was the same API. I have just changed separation of our custom enumerator to use an `#if`.